src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs

```diff
@@ -139,7 +141,7 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
 
             // Semicolons in an empty for statement.  i.e.   for(;;)
             if (previousParentKind == SyntaxKind.ForStatement
-                && this.IsEmptyForStatement((ForStatementSyntax)previousToken.Parent))
+                && this.IsEmptyForStatement((ForStatementSyntax)previousToken.Parent!))
```

Comment:
alternatives:

1. allow null for `IsEmptyForStatement(...)`
2. Should we have a `RequiredParent()`?  or is it too much of a mouthful?

Reply:
The formatter is weirdly perf-sensitive, so I'm trying to avoid adding new branches.